app/controllers/spree/api/v2/storefront/passwords_controller.rb

```diff
@@ -0,0 +1,33 @@
+module Spree
+  module Api
+    module V2
+      module Storefront
+        class PasswordsController < ::Devise::PasswordsController
+          def create
+            user = Spree.user_class.find_by(email: params[:user][:email])
+
+            if user&.send_reset_password_instructions
+              head :ok
+            else
+              head :not_found
+            end
+          end
+
+          def update
+            user = Spree.user_class.reset_password_by_token(
+              password: params[:user][:password],
+              password_confirmation: params[:user][:password_confirmation],
+              reset_password_token: params[:id]
+            )
+
+            if user.errors.empty?
+              head :ok
+            else
+              head :unprocessable_entity
```

Comment:
@pmilewski we need to return that error message

```suggestion
              json: { error: user.errors.full_messages.to_sentence }
```